File: FunctionAndArrays/rotateArray.java
import java.io.*;
import java.util.*;

public class Main {
    public static void display(int[] a) {
        StringBuilder sb = new StringBuilder();

        for (int val: a) {
            sb.append(val + " ");
        }
        System.out.println(sb);
    }
    public static void reverse(int[] arr,int i,int j) {
        while(i<j){
            int temp=arr[i];
            arr[i]=arr[j];
            arr[j]=temp;
            
            i++;
            j--;
        }
    }
    public static void rotate(int[] arr, int k) {
        // write your code here
        int n= arr.length;
        k = (k % n + n ) % n;
        reverse(arr,0,n-1);
        reverse(arr,0,k-1);
        reverse(arr,k, n-1);
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = Integer.parseInt(br.readLine());
        }
        int k = Integer.parseInt(br.readLine());

        rotate(a, k);
        display(a);
    }

}

File: FunctionAndArrays/rotate.Array.cpp
#include<iostream>
#include<vector>

using namespace std;

void reverse(vector<int>& arr,int i,int j) {
    while(i<j){
        int temp=arr[i];
        arr[i]=arr[j];
        arr[j]=temp;
            
        i++;
        j--;
    }
}
void rotate(vector<int>& arr, int k) {
    int n= arr.size();
    k = (k % n + n ) % n;
    reverse(arr,0,n-1);
    reverse(arr,0,k-1);
    reverse(arr,k, n-1);
    }
    
int main(int argc, char **argv){
    
    int n, k; cin>>n;
    vector < int > arr(n);
    for(int i; i<n; i++) cin>>arr[i];
    cin>>k;
    rotate(arr,k);
    for(int ele : arr) cout<<ele<<" ";
    
    return 0;
}
